File: data/gpt-4o/readable_format_HPEC/cpp/file257.cpp
#include <cmath>
#define DATA_TYPE double
#define _PB_N n

void kernel_cholesky(int n, DATA_TYPE* p, DATA_TYPE** a) {
  DATA_TYPE x;
  for (int i = 0; i < _PB_N; ++i) {
    x = a[i][i];
    for (int j = 0; j < i; ++j) {
      x = x - a[j][i] * a[j][i];
    }
    p[i] = 1.0 / sqrt(x);
    for (int j = i + 1; j < _PB_N; ++j) {
      x = a[j][i];
      for (int k = 0; k < i; ++k) {
        x = x - (a[k][j] * a[k][i]);
      }
      a[i][j] = x * p[i];
    }
  }
}

File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file257.f90
subroutine kernel_cholesky(n, p, a)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n) :: p
DATA_TYPE :: x
integer :: n
integer :: i, j, k

!$pragma scop
do i = 1, _PB_N
x = a(i, i)
do j = 1, i - 1
x = x - a(j, i) * a(j, i)
end do
p(i) = 1.0D0 / sqrt(x)
do j = i + 1, _PB_N
x = a(j, i)
do k = 1, i - 1
x = x - (a(k, j) * a(k, i))
end do
a(i, j) = x * p(i)
end do
end do
!$pragma endscop
end subroutine
